### PR TITLE
Issue #7

### PR DESCRIPTION
**Pull request by fadlinurhasan**
Adjust code to stored detail information for image source and remote

### CODE REVIEW

File: ext/client.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ type Client interface {
 	GetNode(string) (*pfmodel.Node, error)
 	GetContainers() (*pfmodel.ContainerList, error)
 	GetContainer(string) (*pfmodel.Container, error)
-	CreateContainer(string, string, string, string) (*pfmodel.Container, error)
+	CreateContainer(string, string, string, string, string, string, string) (*pfmodel.Container, error)
```
Comment (giosakti) [inline ext/client.go]: this is code smell, need to be extracted into data structure, for example: source
ideally CreateContainer should only have (hostname string, source) as attributes

File: ext/client.go [hunk 2]
```diff
@@ -199,9 +199,12 @@ func (c *client) CreateContainer(hostname string, image_alias string, image_serv
...
 	form := url.Values{}
 	form.Set("container[hostname]", hostname)
-	form.Set("container[image_alias]", image_alias)
-	form.Set("container[image_server]", image_server)
-	form.Set("container[image_protocol]", image_protocol)
+	form.Set("container[source_type]", source_type)
+	form.Set("container[source_alias]", source_alias)
+	form.Set("container[source_certificate]", source_certificate)
+	form.Set("container[source_mode]", source_mode)
+	form.Set("container[remote_server]", remote_server)
```
Comment (giosakti) [inline ext/client.go]: do we have these params (remote_server, remote_protocol) accepted on mono?
i thought it should be source[remote[name]]

File: ext/container.go [hunk 3]
```diff
@@ -12,13 +12,24 @@ type ContainerRes struct {
...
+type Source struct {
+	Type        string `json:"source_type"`
+	Alias       string `json:"alias"`
+	Certificate string `json:"certificate"`
+	Mode        string `json:"mode"`
+	Remote      Remote `json:"remote"`
+}
+
+type Remote struct {
+	Server   string `json:"server"`
```
Comment (giosakti) [inline ext/container.go]: remote also have identifier attribute `name`

File: pfmodel/container.go [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
-	Ipaddress     string
-	ImageAlias    string
-	ImageServer   string
-	ImageProtocol string
-	NodeHostname  string
-	Status        string
+	Hostname     string
+	Ipaddress    string
+	NodeHostname string
+	Type         string
```
Comment (giosakti) [inline pfmodel/container.go]: source and remote can be encapsulated

**Review by giosakti — CHANGES_REQUESTED** [review]

File: ext/client.go (hunk 1 above)
Comment (fadlinurhasan) [reply ext/client.go]: noted.

File: ext/client.go (hunk 2 above)
Comment (fadlinurhasan) [reply ext/client.go]: Done. please check Mas.

File: ext/container.go (hunk 3 above)
Comment (fadlinurhasan) [reply ext/container.go]: Done. please check Mas.

File: pfmodel/container.go (hunk 4 above)
Comment (fadlinurhasan) [reply pfmodel/container.go]: Done. please check Mas.

**Pull request merged**